azurelinuxagent/ga/exthandlers.py

```diff
@@ -871,6 +917,34 @@ def collect_ext_status(self, ext):
 
         return ext_status
     
+    def is_ext_handling_complete(self, ext):
+        status = self.collect_ext_status(ext)
+        
+        # when there is no status file, it is incomplete
+        if status is None:
+            return (False, status)
+
+        # If not in terminal state, it is incomplete
+        if status.status not in EXTENSION_TERMINAL_STATUSES:
+            return (False, status.status)
+
+        # Main status is terminal.
+        # However, if there is any substatus in terminal state, it it incomplete
+        for substatus in status.substatusList:
+            if substatus.status not in EXTENSION_TERMINAL_STATUSES:
```

Comment:
This check looks incorrect... an extension may report a warning in a substatus and success in the status. This check would incorrectly return that the extension has not completed.